test/es-module/test-esm-loader-http-imports.mjs

```diff
@@ -53,20 +85,14 @@ const {
     ]
   );
 
-  let stderr = '';
-  let stdout = '';
-
-  child.stderr.setEncoding('utf8');
-  child.stderr.on('data', (data) => stderr += data);
-  child.stdout.setEncoding('utf8');
-  child.stdout.on('data', (data) => stdout += data);
+  const stderr = (await child.stderr.toArray()).join('');
+  const stdout = (await child.stdout.toArray()).join('');
+  const [code, signal] = await once(child, 'close');
 
-  child.on('close', mustCall((code, signal) => {
-    strictEqual(stderr, '');
-    strictEqual(stdout, '[Module: null prototype] { sum: [Function: sum] }\n');
-    strictEqual(code, 0);
-    strictEqual(signal, null);
+  strictEqual(stderr, '');
+  strictEqual(stdout.replace(/\r\n/g, '\n'), '[Module: null prototype] { sum: [Function: sum] }\n');
+  strictEqual(code, 0);
+  strictEqual(signal, null);
 
-    server.close();
-  }));
+  server.close();
```

Comment:
Is there a reason you changed the style of this test? In https://github.com/nodejs/node/pull/43784 @JakobJingleheimer is working on refactoring tests like this to use a helper for spawning child processes. I would leave the old patterns alone, and follow it for new tests you’re adding, so that it’s easier to refactor to use the new helper once it’s ready.